### DISCUSSION TITLE
How to turn on double-click to open folders in elemntaryOS 6?

### DISCUSSION THREAD

**Discussion by sryze** [elementary/files#1877] [Q&A] [answered]
The solutions listed at https://elementaryos.stackexchange.com/questions/38/configure-files-to-use-double-click didn't work unfortunately... :disappointed: 
Maybe this should be included somewhere in system settings? I don't understand why it's not there to be honest

**Comment by jeremypw** [comment, marked as answer]
Double and single click modes have been replaced by a mixed mode (single click on folder navigates, single click on file selects) - the previous two modes are no longer supported and cannot be re-enabled without changing the code.  For further discussion arising see https://github.com/elementary/files/discussions/1729

**Reply by sryze** [reply]
OK, another question then: how are you supposed to select a single folder for copying in mixed mode?

**Reply by sryze** [reply]
Never mind, looks like it can be done with Ctrl+Click

**Reply by jeremypw** [reply]
Yes there are several other ways.  e.g. click on helper (tick) emblem, or use rubberbanding (click-drag anywhere except on the icon)

**Reply by sryze** [reply]
Are there any plans to bring double-click back? I really got used to it... Also, some programs use double-click as well, such as Firefox's file dialog or file-roller, that makes it inconsistent right now.

**Reply by jeremypw** [reply]
There are no plans to bring double-click back. The consistency point is valid but that is more likely to be addressed by writing a FileChooser  portal that behaves similarly to Files.  At the moment we are only using a Gtk portal with a modified pathbar.

**Reply by junglira** [reply]
I do not understand this very extreme need to be different with file management. Is there a reason behind it other than to be different from everyone?

Why would having two totally contrasting and inconsistent selection methods ever be simpler than having the uniform and standard double click behavior for files and folders? That is an unnecessary complication to what should be a very simple task. There is a reason it exists this way universally across Mac, Windows, and just about any other Unix/Linux desktop. Why is forcing users to install a new file manager the only option to get around this behavior?

Needing to undo decades of computing standards is not "thoughtful" nor "easy", which goes against elementaryOS's philosophy. At the bare minimum leave the terminal option to resolve the inconsistent selection behavior.

**Comment by jeremypw** [comment]
I think the rationale is that Files is mainly used for navigating through the filesystem in order to locate and operate on files.  To do this single-click navigation is optimum for folders and single-click selection is optimum for files.  Selecting folders is still easy if you do need to but I certainly do this far less often than navigating.

It may be helpful to look at what workflows require selecting folders and see whether they can be made easier in Files or in some other way.

**Reply by junglira** [reply]
All file managers are used for navigating through filesystems to operate on files, so I'm not sure what that explains that is specific to Files. Unless you are implying you shouldn't be using folder in Files, which I sincerely hope that isn't the case. And it might be a good reminder that folders are indeed files in Linux, as that seems to be missed here.

Also having two inconsistent selection methods for files and folders is definitely not optimal or easy. It still isn't clear why making a user think in two different selection mindsets is easier than allowing them to just use the same techniques uniformly across both. It is a decades old standard, why would you want to add a complication to this very fundamental aspect when the OS is aiming to be friendly and usable?

I've seen a strange 'argument' in that it is supposed to make it similar to navigating the web. At the risk of pointing out the extreme obvious flaw in this mindset, you don't actually do file management on web links. You click web links and that is it. Folders you do file management on and often you need to create, move to other locations, delete, select. There is a very distinct difference and why that mindset and trying to apply the same usage patterns makes no sense.

I can only see this abrasive head scratcher of a decision being hurdle to elementary. Forcing users to re-learn how to do basic file operations is not a good way to draw people from other distros or OSs. This is too much of a fundamental component to be forcing changes onto users. Please do not make this mistake. At the very least follow the Linux philosophy and allow users to customize this behavior. That in turn will follow elementary's own philosophy in making the computer accessible and thoughtful as well.

**Comment by jeremypw** [comment]
The fact that most mainstream filemanagers offer both single-click to open and double-click to open tends to suggest that neither is clearly superior.  I found both modes irritating in one way or another.  Elementary philosophy, for better or worse, is to offer the single optimum (in their opinion) way of doing things.  The argument that mixed mode is confusing to users well used to one or other of the existing options is clearly true, at least initially, but I must assume that all the people who actually develop elementary can be regarded as "power users" and use Files for a multitude of purposes.  No issues have been raised internally about moving to mixed mode (which was discussed and trialled before release).  It is very unlikely so soon after release that the decision would be reversed but I guess it cannot be ruled out that, should vast numbers of users fail to adapt to it, things could change.   I am sure that some users will install other filemanagers just as they no doubt install other music players, photomanagers, video players etc and that is fine.

**Reply by junglira** [reply]
> The fact that most mainstream filemanagers offer both

Right there! This is another major key. They offer both. And there is a very unusual and very extreme distaste here to give users the option in Files to chose one way or the other. I haven't seen another file manager force a user one way or the other without options. And if you are intending to force a user with a single setting, why would go with the least popular option of introducing two very different operation methods for files and folders instead of the universal double click? It doesn't really add up. As I said before, and seemingly you agree with too from that statement, just offer both selection methods as an option. I'm arguing for the case where elementary wants to remove user option and that double click should then be the default.

> I must assume that all the people who actually develop elementary can be regarded as "power users"

The argument for power users doesn't really work because any power user would be almost guaranteed to have used double click on another OS or distro to become a power user in the first place. Then they would need to unlearn years of experience to do simple file management (such as a live case with me). And if there are power users who do want the inconsistent clicking method, they will be able to enable it. It isn't a good default option for trying to bring people over when 99% of them don't use this control scheme. And if it's a "true" power user who uses terminal exclusively for most operations, it's extremely unlikely they'd be using elementary in the first place. They would need much more control of the computer than elementary allows for without extensive work and customizations, and in which case other distros would be more appropriate.

Additionally, I was originally attracted to elementary because it was supposed to be "easier". I should just be able to install it and work on my code without problems. Instead there's basic file management problems, I have to reinstall the OS to upgrade.... just a lot of surprising hassle that I haven't found in other distros.

>  I am sure that some users will install other filemanagers just as they no doubt install other music players, photomanagers, video players

I hope you see there is a major difference between a specific application like a  music player, and one that is a primary face of the OS - the file manager. This isn't just any piece of software. Why would you not want users to primarily use the OS's own file manager and why would you encourage them to explore elsewhere? In that case, why develop Files at all and just use one already available? Or why not give the users the choice of which comes during the install if you encourage it? That just raises so many questions.

Mostly I'm trying to understand what rational is behind this. There are numerous reasons not to take this approach but I haven't seen any real reason supporting it. Why make things complicated?

**Reply by sryze** [reply]
> Additionally, I was originally attracted to elementary because it was supposed to be "easier". I should just be able to install it and work on my code without problems. Instead there's basic file management problems, I have to reinstall the OS to upgrade.... just a lot of surprising hassle that I haven't found in other distros.

+1, I like elementary for its clean design and ease of use, but there are just some questionable choices made by the devs that I don't understand.

For example, after installing elementaryOS, I need to also install a [third-party tool](https://github.com/Lafydev/wingpanel-indicator-ayatana) to be able to see Skype, Nextcloud, Dropbox and other apps in the system tray panel. Every other Linux distro has this built-in, but in elementaryOS it has been removed for some odd reason.

**Reply by junglira** [reply]
Exactly my thoughts @sryze. It's got a great design and *sometimes* is easy to use, but there are such... very, *very* puzzling and troubling choices being made lately, even besides the issues we listed. Sadly, most of these choices only seem to boil down to "Well, we kinda just felt like it. Deal with it." and don't take the users into account at all.

I really want them to succeed, but they seem to be actively trying to add hurdles for sake of.. adding hurdles? I can't see those decisions giving them a particularly bright future sadly.

I wish they could just answer this one question. The very first sentence on the website is "The thoughtful, capable, and ethical replacement for Windows and macOS". So they are clearly trying to bring people over from macOS and Windows - both of which use a unified double click method for operating on files and folders. Why on earth would you want to make it complicated for the very people you are bringing over by making it so they can't use the file manager effectively? It's so hard to understand why they would force the inconsistent one and two click method instead of keeping it double click to help transition people over.

**Comment by jeremypw** [comment]
As I said before I am struggling to identify what the need to *regularly* select and manipulate folders is. I hardly ever do this.  If there is such a need it probably points to a deficiency in one of the other more specific apps e.g. Code.  Personally, I am not sure that elementaryos should consider itself as a "replacement" for anything - it is more accurately an "alternative". 

> Forcing users to re-learn how to do basic file operations is not a good way to draw people from other distros or OSs. 

I think the intention is more to get to a point where "basic file operations" are rarely needed to be done by the user with a file manager.

> one that is a primary face of the OS - the file manager.

I do not think that elementary would agree that the file manager is or should be the primary face of an OS.

**Reply by junglira** [reply]
> As I said before I am struggling to identify what the need to regularly select and manipulate folders is.

I think the problem is you are only thinking of yourself. Just because you don't do it, doesn't mean most users don't (which is very hard to believe you don't do regular file management). I feel I have to ask, Files is intended to be used more than by just you, right? 

There are multiple users complaining about wanting double click, even beyond the two of us. elementaryOS is trying to bring over users from OSs that do use double click and you _should_ want to make that transition easy. I think all of that speaks for itself and is abundantly clear in why double click should be used if you insist on giving no options to change it. I'm very confused that any one part of this is unclear.

You also can't say elementaryOS isn't a replacement - it literally says it in plain text directly on the website that elementaryOS is, in fact, a replacement. You can see this for yourself as the very first line on the website: https://elementary.io

> I think the intention is more to get to a point where "basic file operations" are rarely needed to be done by the user with a file manager.

Just to be clear, elementaryOS is designed for a computer and not a tablet, right? On a computer, people do file management excessively and it's very unlikely to change soon. Especially the "power users" you claimed this OS is for. I'm not sure how you've used a computer and not done file management. Do you download files, code, or do art? There are endless reasons for file management. 

And even if somehow you don't need file management regularly, it is still not a good argument for making it harder to use the file manager.

> I do not think that elementary would agree that the file manager is or should be the primary face of an OS.

It's not that you can decide this or not, that's just how it is. It's the main interface to using the OS and is the face people will give to it. Same with Mac's Finder or Window's Explorer, which are iconic faces to both OSs. Please tell me you understand that. It's a _very_ critical piece as to why file managers are so important and can easily make or break a system when people can't do basic operations with it well.

I noticed my main question still wasn't answered. As stated directly on elementaryOS's website "The thoughtful, capable, and ethical **replacement** for Windows and macOS". The clear goal of elementaryOS is to bring people over from Mac and Windows. You can deny that if you like, but it it is in plain text as the mission statement on the website. And since elementaryOS is supposed to bring people over from the two OSs, why would you insist to make it harder for them? Both of those OSs use double click. The people you are trying to bring over will expect double clicking. You are actively making it harder for them to switch. And my question is, why?

**Reply by sryze** [reply]
> As I said before I am struggling to identify what the need to regularly select and manipulate folders is. I hardly ever do this. 

Hm, is this why Files is not pinned to the dock by default? It's a little odd to not have the file manager there after installing elementaryOS. If I was a new user barely familiar with Linux, I would probably have to google how to open files on my disk :smiley:

**Reply by Marukesu** [reply]
> Hm, is this why Files is not pinned to the dock by default? It's a little odd to not have the file manager there after installing elementaryOS. If I was a new user barely familiar with Linux, I would probably have to google how to open files on my disk :smiley:

I believe the idea is that the user should use the appropriate application to manage the files, for example, code files and project folders should be handled in code, images in photos, etc...

> I noticed my main question still wasn't answered. As stated directly on elementaryOS's website "The thoughtful, capable, and ethical replacement for Windows and macOS". The clear goal of elementaryOS is to bring people over from Mac and Windows. You can deny that if you like, but it it is in plain text as the mission statement on the website. And since elementaryOS is supposed to bring people over from the two OSs, why would you insist to make it harder for them? Both of those OSs use double click. The people you are trying to bring over will expect double clicking. You are actively making it harder for them to switch. And my question is, why?

IMHO, changing the default behaviour to double click would frustrate the current user base of elementary that get used to single click in the last 10 years. the current users are than important as new users.

I agree with Jeremy that "alternative" is a better way to describe elementary than "replacement", elementary has they own philosophy and things can be different from what others OSes/distros do.

**Reply by junglira** [reply]
> I believe the idea is that the user should use the appropriate application to manage the files, for example, code files and project folders should be handled in code, images in photos, etc...

I cannot believe this is the consistent reason to allow Files to use bad practices. Why do you think file mangers exist in the first place on every OS if they aren't supposed to be the primary way to manage files? There are also a number of problems with this approach.
1. You will have a VERY inconsistent user experience. Some applications may offer clunky/bad interfaces, some may not give you enough power to do the management you need, and some may have both issues. And some may be much better than others, which leads to problem 2....
2. This will lead to very terrible use practices. For an example, you could wind up opening the Photos app to manage Music files because Photos app handles file management better.
3. You do not want to have an app doing too much. Each app should be dedicated and have distinct use. It should not be Photo manager + also the File Manager.
4. Redundancy will be at an extreme high with every app needing to replicate entire file manager functionality.
5. Multiple different apps trying to all pretend to be the file manager at the same time... surely you can see the possible race conditions and issues resulting from this.

Also, are you supposed to add a file manager into the browser now so you can move files after downloading them? To simple sound recoding software? Inside every game to adjust save files? Please, be reasonable.

> IMHO, changing the default behaviour to double click would frustrate the current user base of elementary that get used to single click in the last 10 years. the current users are than important as new users.

I think this is only more of glaring reason why you MUST allow the option to choose between sensical double click and the inconsistent single + double click. You had dug your grave yourselves, so you need to resolve the problem that was created from your own decisions. You have a user base that expects this - and you will deter users away from joining because you decided to go with this unusual approach. The only solution if you refuse to change is to allow the users to set this.

Besides that, isn't mixed mode a new feature? How are your previous users "used to this for 10 years" if it's brand new? I think you may be mistaken. And either way, it shows you'll change things on your old users anyway - so may as well help the new users in that case. Plus not all old users used single click. I've used elementary for years and would set it to double click. I'm sure there are many old users who've done that, and you've robbed them of the ability to use the file manager effectively now that you went out of the way to remove that.

> I agree with Jeremy that "alternative" is a better way to describe elementary than "replacement", elementary has they own philosophy and things can be different from what others OSes/distros do.

Please go to elementary's own website for clarification on this. https://elementary.io. It says in plain text "replacement". You can believe all you want, but it does not change the fact it is indeed sold as a replacement OS. I'm very alarmed the people in charge of the File Manager do not even know the mission statement for the OS they are creating it for.

And regardless if it is a replacement or alternative OS, that, again, does not give a good reason to give a bad user experience. "The thoughtful, capable, and ethical **replacement** for Windows and macOS". You are trying to bring over macOS and Windows users. They expect double click. Why are you insisting to make it hard to increase your user base?

**Comment by jeremypw** [comment]
It is self-evident that elementary cannnot be a slot-in UX replacement for *all* Windows and MacOS and other Linux OSes. It replaces (much of) the functionality of those OSes but as it only has one UX it will differ in details of how the functions are carried out. There are other OSes that specialize in emulating traditional Windows and MacOS UX. The resources available to elementary compared to Windows and MacOS differ by several orders of magnitude.

Most of the examples of file manager use given above are about operations on files not folders and can mostly be carried out via the filechooser interface so the apps do not need general file management capability.  Photos and Music organize their content via libraries and databases and display the content in a way suitable for the particular type. Details of the filesystem do not (or should not) generally need to be exposed to the user.

I have opened a new discussion about ways in which mixed-mode UX could be improved (other than scrapping it or adding a double-click option) as this discussion is beginning to generate more heat than light.

**Comment by junglira** [comment]
> It is self-evident that elementary cannnot be a slot-in UX replacement for all Windows and MacOS and other Linux OSes.

Perhaps you should edit the website then to not state that it is indeed a replacement, no? Why say replacement, if it is not a replacement? Very misleading.

> generate more heat than light.

I think this discussion has generated quite a lot of light and evidence to use double click, but you Marukesu seem to prefer ignoring these facts and instead try to drum up random excuses each time to allow the mixed mode. I'm willing to bet you might even be running out of these excuses with the sudden change in direction and saying nothing against my proofs for it.

So again, if the primary focus is stated on the website to replace Windows and Mac, and you are trying to bring people over from Windows and Mac that expect double click, why do you *want* to make the transition harder for them? And it is clear with the new mixed mode that this isn't "to protect old users" as mixed mode replaces the previous mode. Not to mention many old users used double click and you are removing that.

So if it's not for new or old users, it clearly seems there is no good reason at all to use mixed mode over double click when double click has many advantages over preference.

**Reply by jeremypw** [reply]
> Perhaps you should edit the website then to not state that it is indeed a replacement, no? Why say replacement, if it is not a replacement? Very misleading.

This is not within my remit. If you raise an issue report at https://github.com/elementary/website/issues then it will be considered.

**Reply by junglira** [reply]
> This is not within my remit. If you raise an issue report at https://github.com/elementary/website/issues then it will be considered.

Hmm. So you are perfectly fine with leaving this 'incorrect' statement on the website despite your firm belief it is false? If you can't be bothered to report it, I can pickup the slack if you wish.

And also thank you for once again ignoring my question. You have at least confirmed my fear that the reason for not using double click is solely for your own (and possibly a very small group's) personal preference exclusively. It has little input or care for users or their benefit.

It was my own mistake to believe long time user input could be at least considered, or that you would care for helping ease in new users, and I'm sorry I wasted time trying to bring these points up to deaf ears.

Good luck with the project.

**Comment by idovecer** [comment]
I also prefer double click for opening folder and files. I can't get used to Files in elementary because it's confusing and illogical.
Single click to open the folder but double click to open the file?!

Please don’t be stubborn and add one option so people can simple turn on double-click or single-click at will.
Linux is an alternative to stubborn OS systems like Windows and OSX, so be better and different.

I hope that one day elementary FILES will have some simple options to configure you preferred file exploring styles.

**Reply by junglira** [reply]
That's exactly what I was trying to explain as well.

But, as sad as it is to say, don't get your hopes up. They made it abundantly clear in this and other discussions that they have no intentions to provide user options, nor will they consider changing it to have a uniform single or double click. And as I uncovered above, it literally benefits nobody but their own preference. Old users are forced to change to this style, as are new users. It's seriously mind boggling the direction they are taking here... well this is one of many unfavorable decisions as of late unfortunately.

Sadly, the only real options you have are to either conform to their every whim, or save yourself the time and effort and find another distro. I wish it wasn't the case, but you can see how my trying to reason with them went. Simply a waste of time.. Find another distro that will respect you having a preference.

**Reply by idovecer** [reply]
I was using ubuntu 10.04 and 12.04, then go to the Lubuntu 14.04, 16.04 and 18.04.

The Lubuntu was perfect, very very fast and there you can be very productive. I changed to Elementary OS Hera because I want something new, I like the design. I will see until 22.04, maybe I will get back to Lubuntu.

I do like lubuntu PCManFM file explorer more than elementary Files and eOS is kinda slow comparing zo super fast Lubuntu. Tried to use PCMan FM or others file explorers in eOS, but Files is kinda "integrated" to system almost like on beggining IE to win.

**About single/double click**, I also use a lot **Filezilla**, and there again is **doubleclick** for opening folders.
Then when you save something for example via **LibreOffice** and you need to change folder, you again must **double click**, not single click anymore.

It is kinda inconsistent.

**Reply by junglira** [reply]
I ended up moving to Pop!_OS myself after my fruitless discussion here and getting a feel for the developer's direction. That and getting fed up with one thing after another with elementary as of late (forced system trashing to upgrade, forced to use inconsistent double and single click, less and less options for every program, somehow worse speeds each upgrade with less user power, and so on). I got tired of having to do extreme changes, hacks, and tweaks to elementary every update in order to make it usable again. Now it's only a fool's journey to do, and you'd be better off doing Linux from scratch for a better return on the time investment.

There is a lot of common ground between Pop and elementary, so I found it to be a very easy transition. Especially after years of trying to stick with (_and donating to_) elementary before being bullied away from it by the devs I once tried to support. Maybe you'll find it to be an easy jump too. I particularly like the ability to turn on and off tiling and the dedicated/integrated graphics easily. In addition it's really nice having options in programs, being encouraged to make the system your own, and not worrying about more freedom being removed from me each update. Though those latter points are features of basically any distro; well, sans elementary of course. Still so incredible to realize basic features like that can be used as highlights.

**Comment by jeremypw** [comment]
There are no plans to revert to a multiple UX design but there is a PR in progress that will increase the target area for single-click selection/double-click opening of folders. You just have to remember to not click on the icon (or, at the moment) filename.  Alternatively holding down the `Alt` key gives you single-click select for folders.  We are open to ideas to improve the current mixed mode UX short of reverting to two different UX options.

**Reply by idovecer** [reply]
OK thank you, for ALT + mouse click tip. It's kinda annoying to press the alt button all the time:
Alt + Left mouse click (just left mouse click is better)
Alt + LEFT on keyboard to go BACK (just backspace is also etter in my opinion)
Anyway, there is a progress to better, thanks I will try to get more familiar to this logic.

In my opinion there is no need to increase single click area, I use CTRL + mouse click for selection or Shift + mouse click.

**Comment by JohnDumpling** [comment]
An option to use double click both for navigation and opening files has been added in one of the recent updates:
![double click](https://github.com/elementary/files/assets/89086803/49b920ee-70c5-4e6f-af77-72954f7528d6)

**Reply by junglira** [reply]
Blast from the past. And it's finally in.

I'm still just blown away how incredibly stubborn they were over this. They fought so, so hard about even _considering_ it, let alone adding it. They came up with philosophical reasonings, their own preferences, how elementary wasn't a desktop replacement, that the then current "users of ten years" would (for some reason) become frustrated adding this as an option, among many other excuses.

I guess we should all just be thankful it's actually in. I'm genuinely and wholeheartedly impressed they were able to overcome their proudness to consider the needs of their users. It's an unmistakably good step forward for elementary. It's just a shame they didn't give this (among other issues) even the slightest consideration back when I used to donate and support them. Entirely my own mistake for believing and lending support then, but a shame nonetheless.

I'm sure users of elementary will greatly appreciate finally having this very basic feature after many years of asking for it. And I genuinely hope they can continue this path forward in overcoming previous issues to improve elementary.

**Reply by BugFixesAndPerformanceImprovements** [reply]
It unfortunately is still single-click only in the file picker (like when opening or saving files).

**Reply by junglira** [reply]
Oh no haha. I guess ignore my compliments and mention of progress. Thanks for that confirmation BugFixesAndPerformanceImprovements. Case in point what I get for having faith in them. I was too excited when I got that random GitHub notification email years later.

**Reply by BugFixesAndPerformanceImprovements** [reply]
You can read more about it in this [issue](https://github.com/elementary/files/issues/2104). It's been 7 months.
